### ISSUE TITLE
[Bug]: Invalid variable renaming in Vs code action

### ISSUE THREAD

**Issue by SasinduDilshara** [ballerina-platform/ballerina-lang#40130]
### Description

Consider following code sample

```
import ballerina/lang.regexp;

public function main() {
    regexp:RegExp _ = re `(?${""}:)`;
    3;
}
```

When we select the create variable code action for second statement, the renaming option contains invalid variable name.

![Screenshot 2023-04-06 at 16 46 21](https://user-images.githubusercontent.com/47134066/230362340-6f7e9007-db6f-4250-9e43-5b1c27f65cf5.png)

### Related area

-> Compilation

**Comment by SasinduDilshara** [comment]
Another issue in variable renaming.
When we select `create variable` code action and rename our variable to `_`, vs code give two choices, first one is keep the variable as `_` or kepp the variable as `'_`.
When we choose `'_`, the code gives compile time error `'_' is a keyword, and may not be used as an identifier(BCE3107)
`

```
public function main() {
    int '_ = 3; // '_' is a keyword, and may not be used as an identifier(BCE3107
}
```

**Comment by mindula** [comment]
related issue https://github.com/ballerina-platform/ballerina-lang/issues/40525

**Comment by mindula** [comment]
Blocked by #41282
